Cargo.toml

```diff
@@ -52,5 +108,18 @@ required-features = ["utils"]
 # Patch for radio-hal to update embedded-hal version
 radio = { git = "https://github.com/ryankurte/rust-radio.git", branch = "master" }
 
+
 # Fix for libusb-sys package discovery
-libusb-sys = { git = "https://github.com/cmsd2/libusb-sys" } 
+#libusb-sys = { git = "https://github.com/cmsd2/libusb-sys" } 
+
+[[example]]
```

Comment:
maybe useful to gate these with an `examples` feature that the device-specific features can depend on too? that way with the wrong target `cargo build --examples` won't fail.

Reply:
I'll try to figure this out.

Reply:
unless it's not available for examples i'd expect it should work the same as the [`util`](https://github.com/rust-iot/rust-radio-sx127x/blob/master/Cargo.toml#L48) feature

Reply:
I can't find where `libusb-sys` is used. I've had it commented out for a long time, but when I uncomment it I get 
```
warning: Patch `libusb-sys v0.2.3 (https://github.com/cmsd2/libusb-sys#096f2478)` was not used in the crate graph
```
and it does not show up in tree.  Do you know where/how it used? Perhaps this is the source of the CI failure on master?

Reply:
While looking more at this I discovered some sections of rust.yml that I had temporarily commented out and forgotten to uncomment. They were not needed for my fork work, but they are needed on upstream/main. Sorry about the omission. Part of it was the section
```
          - target: armv7-unknown-linux-gnueabihf 
            os: ubuntu-latest
            output: sx127x-util
            use_cross: false
            experimental: true
            args: --features util,driver-pal/hal-cp2130,driver-pal/hal-linux

```
which is one place the `libusb` is used. So I am getting close to being able to at least reproduce the error. (Also sorry this is adding more commits.)

Reply:
I am not having any luck getting the `armv7-unknown-linux-gnueabihf` target to build. Changing `libusb-1.0` to `libusb-1.0-0` in `rust.yml` gets to at least finding the library, but still errors with

```
dpkg: error: need an action option
```

In the Cargo.toml the link for `libusb-sys` looks like it is pretty old so I replaced

```
libusb-sys = { git = "https://github.com/cmsd2/libusb-sys" } 
```
with

```
 libusb-sys = {version ="0.2", optional = true}
```
as suggested at https://github.com/dcuddeback/libusb-sys. That change fixes the `build-util (x86_64-unknown-linux-gnu` problems but a `armv7-unknown-linux-gnueabihf` problem  remains.

I am a bit uncomfortable with these adjustments because libusb is not used on any platform I have been testing, since it uses std. Also I'm not sure why the "fix" version was installed in the first place.

Possibly I am not understanding what you mean by "gated". In  Cargo.toml I now  have `libusb-sys` as above. I think that works appropriately. The remaining problem seems to have to do with the `rust.yml` or `apt install` syntax. But maybe you know of a command  where it does not gate properly?

Reply:
I've put in an `examples = []` feature and use it as a gate.  I think there are no longer problems related to `libusb-sys` in cargo usage and workflow. There is the question of whether I should have upgraded `libusb-sys` to `version ="0.2"` from the git fork you were using, because I am not sure what the git version was fixing.

While doing this I was also trying to get the CI workflow correct, and also `cargo test` and `cargo build examples` working. The last I think is working correctly now. The first is improved but the job
`build-util (armv7-unknown-linux-gnueabihf,... ` is still failing. (This is a long standing problem in your CI I think.) It gives

```
  sudo apt-get install libusb-1.0-0:armhf libusb-1.0-0-dev:armhf 
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    GITHUB_REPOSITORY_SLUG: pdgilbert-rust-radio-sx127x
    GITHUB_REPOSITORY_OWNER: pdgilbert
    GITHUB_REPOSITORY_OWNER_SLUG: pdgilbert
    GITHUB_REPOSITORY_NAME: rust-radio-sx127x
    GITHUB_REPOSITORY_NAME_SLUG: rust-radio-sx127x
    GITHUB_REF_SLUG: refs-heads-master
    GITHUB_REF_NAME: master
    GITHUB_REF_NAME_SLUG: master
    GITHUB_SHA_SHORT: e68cbda4
dpkg: error: need an action option
```
I tried `apt-get` in place of `apt`, and many other things but no luck. Googling suggests a faulty cut and paste of a web "-" character, but I could not find that. The command does work on Raspberry Pi, so I suspect an emulator problem or some faulty translation in github Workflow. 

In `cargo test` there is a problem

```
error[E0004]: non-exhaustive patterns: type `&mut HalSpi` is non-empty
   --> /home/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/driver-pal-0.8.0-alpha.2/src/hal/mod.rs:170:15
    |
145 | / pub enum HalSpi {
146 | |     #[cfg(all(feature = "hal-linux", target_os = "linux"))]
147 | |     Linux(linux_embedded_hal::Spidev),
148 | |     #[cfg(feature = "hal-cp2130")]
149 | |     Cp2130(driver_cp2130::Spi),
150 | | }
    | |_- `HalSpi` defined here
...
170 |           match self {
    |                 ^^^^
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
    = note: the matched value is of type `&mut HalSpi`
```
I think it is because the `no_std` `spi ` parts are not found. This does not seem to affect builds, but I guess some lint does not like it.  I don't know how to deal with this. Any suggestions, or should it be ignored?

I have now squashed the last bunch of commits.